joomla/joomla-cms
# Support for ACL permissions per module in com_modules

This pull request allows to set permissions per module in the Joomla Module Manager. So besides configuring the com_modules general permissions you can configure the permissions per module. 

You can use this for example to: 
- allow access to com_modules, but only editing a selection of modules
- allow access to com_modules and edit all modules expect one (or more)
- disable state changes for certain modules
- etc..
## Notes

This patch doesn't include the SQL changes for all sample data sets yet. I am working on that but would suggest to review the code meanwhile.
## JoomlaCode Tracker Item

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=28638
## Testers welcome!

Testers are much welcome, let me know if you have any questions / feedback. Thanks in advance!
## Test Instructions

For those interested in testing (thank you!) I have created testing instructions. If you have a fresh Joomla installation with the ACL permissions per module patch applied I would suggest the following steps:
1. Go to the User Manager and create a new test user that is assigned to the "Manager" group
2. Start a different browser and login with your test user (by using a different browser for your test users the sessions with your own account are not mixed-up)
3. Notice that your test user can't browse the Module Manager in the Menu.
4. Go to the Module Manager with your admin account, open the Options and set the "Access Administration Interface" permission for the "Manager" group to "Allowed".
5. Check that your test user is now able to browse to the Module Manager.
6. Confirm that your test user is now able to edit and save the listed modules.
7. Open one of the modules with your admin account, and set all actions for the "Manager" group to "Denied".
8. Confirm that your test user is no longer able to edit the module you just changed the permissions for.
9. Open another module with your admin account, and set only the "Edit State" action for the "Manager" group to "Denied".
10. Confirm that your test user is no longer able to changing the state of this module (publish/unpublish), but still able to change the title for example.
11. Open another module with your admin account, and set only the "Edit" action for the "Manager" group to "Denied".
12. Confirm that your test user is no longer able to edit the module but able to publish/unpublish the module.

Above are several examples listed of tests you can perform on this new feature. Please go ahead and try other permission setting combinations as well. Like: not allowing the create action in the Module Manager settings, and only the actions for a couple modules instead.

Thanks, Sander
